tests/templates/test_layers.py

```diff
@@ -477,6 +478,43 @@ def test_random_layer_weights(self, n_subsystems, tol):
         params_flat = [item for p in params for item in p]
         assert np.allclose(weights.flatten(), params_flat, atol=tol)
 
+    def test_tensor_unwrapped_well(self, monkeypatch):
+        """Test that if random_layer() receives a one element PennyLane tensor,
+        then it is unwrapped successfully.
+
+        The test involves using RandomLayers, which then calls random_layer
+        internally. Eventually each gate used by random_layer receives a single
+        scalar.
+        """
+        dev = qml.device("default.qubit", wires=4)
+
+        lst = []
+
+        # Mock function that accumulates gate parameters
+        mock_func = lambda par, wires: lst.append(par)
+
+        with monkeypatch.context() as m:
+
+            # Mock the gates used in RandomLayers
+            m.setattr(pennylane.templates.layers.random, "RX", mock_func)
+            m.setattr(pennylane.templates.layers.random, "RY", mock_func)
+            m.setattr(pennylane.templates.layers.random, "RZ", mock_func)
+
+            @qml.qnode(dev)
+            def circuit(phi=None):
```

Comment:
How come when the signature is `phi=None` and `phi` is passed a tensor, it's still a tensor, while if the signature is an argument `phi`, it's somehow converted into a `np.ndarray`? :thinking: 

This made me a bit confused.

Reply:
Oh, it's because if `phi=None`, then `phi` is non-differentiable. Non-differentiable parameters are not "processed" in any way.

By not defining a default value for `phi`, we are making it differentiable and `Variable` objects are created with the values that were passed. The Sequence specified is then converted into a numpy array in the `BaseQNode._make_variables` method.